mobile/src/main/java/ch/epfl/sdp/mobile/application/authentication/AuthenticatedUser.kt

```diff
@@ -55,14 +57,32 @@ class AuthenticatedUser(
     }
   }
 
+  /**
+   * Follows the given [Profile] by updating the list of followers of the followed profile with the
+   * uid of the current user.
+   *
+   * @param followed the [Profile] to follow.
+   */
+  suspend fun follow(followed: Profile) {
+    firestore.collection("users").document(followed.uid).update {
+      arrayUnion("followers", user.uid)
+    }
+  }
+
   /** Signs this user out of the [AuthenticationFacade]. */
   suspend fun signOut() {
     auth.signOut()
   }
 
-  /** Returns a [Flow] of the [Profile]s which are currently followed by this user. */
+  /**
+   * Returns a [Flow] of the [Profile]s which are currently followed by this user by going through
+   * all users in the database and checking which ones have the uid of that user in their list of
+   * follower unique identifiers.
```

Comment:
You should be cautious not to dive into too many implementation details in documentation.